src/options/dir_action.rs

```diff
@@ -12,31 +12,31 @@ impl DirAction {
     /// There are three possible actions, and they overlap somewhat: the
     /// `--tree` flag is another form of recursion, so those two are allowed
     /// to both be present, but the `--list-dirs` flag is used separately.
-    pub fn deduce(matches: &MatchedFlags<'_>, can_tree: bool) -> Result<Self, OptionsError> {
-        let recurse = matches.has(&flags::RECURSE)?;
-        let as_file = matches.has(&flags::LIST_DIRS)?;
-        let tree    = matches.has(&flags::TREE)?;
+    pub fn deduce(matches: &Opts, can_tree: bool, strictness: bool) -> Result<Self, OptionsError> {
+        let recurse = matches.recurse > 0;
+        let as_file = matches.list_dirs >0;
+        let tree    = matches.tree > 0;
 
-        if matches.is_strict() {
+        if strictness {
             // Early check for --level when it wouldn’t do anything
-            if ! recurse && ! tree && matches.count(&flags::LEVEL) > 0 {
-                return Err(OptionsError::Useless2(&flags::LEVEL, &flags::RECURSE, &flags::TREE));
+            if ! recurse && ! tree && matches.level.is_some() {
+                return Err(OptionsError::Useless2("--level".to_string(), "--recurce".to_string(), "--tree".to_string()));
             }
             else if recurse && as_file {
-                return Err(OptionsError::Conflict(&flags::RECURSE, &flags::LIST_DIRS));
+                return Err(OptionsError::Conflict("--recurce".to_string(), "--list-dirs".to_string()));
```

Comment:
Doesn't clap have own errors for this that we could throw? But again, more something for follow-up and fine for now.

Reply:
No not on those cases

Reply:
Ok, haven't tried something like this before, just hoped clap had some own conflict error, but nothing we need to solve right now.